tools/clang/test/SemaHLSL/getattribute-at-vertex.legal.hlsl

```diff
@@ -0,0 +1,41 @@
+// RUN: %dxc -T ps_6_2 -E main %s -verify
+
+// expected-no-diagnostics
+struct S1 {
+  nointerpolation float3 f1;
```

Comment:
I think there are missing cases where a struct field is marked nointerpolation and the field is passed to a function.  Then if you create a temporary of that structure in main and pass that field to a function.  What happens in these cases?

Reply:
Do you mean this case:

```hlsl
struct S6 {
  nointerpolation uint f1;
};

float4 main(S6 s6 : G) : SV_Target {
  S6 s6_copy = s6;
  float v8 = GetAttributeAtVertex(s6_copy.f1, 0);
  
  
  S6 s6_copy_2;
  s6_copy_2.f1 = s6.f1;
  float v8 = GetAttributeAtVertex(s6_copy_2.f1, 0);
}
```

If so, I can add those 2 cases. They pass the check as legal, but on the SPIR-V side, they crash (but that's another story 🙃)

Reply:
You don't seem to have the common, legal case of passing `s6.f1` directly to a function that uses `GetAttributeAtVertex`, unless that was covered by some other existing test case I didn't see.  The temporary variable case was just an extension of this to see how it's handled.

Reply:
Right, added one line to check this.